openhands/utils/http_session.py

```diff
@@ -15,13 +17,25 @@ class HttpSession:
 
     session: requests.Session | None = field(default_factory=requests.Session)
 
-    def __getattr__(self, name):
+    def __getattr__(self, name: str) -> Any:
         if self.session is None:
             logger.error(
                 'Session is being used after close!', stack_info=True, exc_info=True
             )
-        return object.__getattribute__(self.session, name)
+            raise RuntimeError('Session is being used after close!')
+        return getattr(self.session, name)
 
-    def close(self):
+    @property
+    def headers(self) -> CaseInsensitiveDict[str]:
```

Comment:
Note: this is necessary because it is used, e.g. in this file:
[openhands/runtime/builder/remote.py](https://github.com/All-Hands-AI/OpenHands/blob/312b9fbfb1c284c87ec57e0998cf3a731f795b70/openhands/runtime/builder/remote.py#L26)